crates/uv-python/src/downloads.rs

```diff
@@ -408,7 +410,18 @@ impl ManagedPythonDownload {
         parent_path: &Path,
         reporter: Option<&dyn Reporter>,
     ) -> Result<DownloadResult, Error> {
-        let url = Url::parse(self.url)?;
+        // Parse the provided URL. If a mirror is set, use it instead.
+        let url = if let Some(mirror) = std::env::var("UV_PYTHON_INSTALL_MIRROR").ok() {
+            let Some(suffix) = self.url.strip_prefix(
+                "https://github.com/indygreg/python-build-standalone/releases/download/",
+            ) else {
+                return Err(Error::Mirror(self.url));
```

Comment:
PyPy downloads will fail this, but I think that's correct.

Reply:
So if you set the mirror and use install PyPy it'll fail?

Reply:
Yes

Reply:
Seems a little weird. I'm not sure what I'd expect though.

Reply:
I could instead _also_ replace the PyPy URL, e.g., strip `"https://downloads.python.org` or `"https://downloads.python.org/pypy`.

Reply:
That might make sense to me, why did you avoid that in the first place?

Reply:
Because the schemas are totally different, i.e., the thing that would come after your mirror URL. For CPython, you need a date, etc. For PyPy, it's just a filename.

Reply:
I see. So we'd need a separate `UV_PYPY_INSTALL_MIRROR` in the future? I'm okay with CPython being implicitly `PYTHON`.

Reply:
(and so on for every additional implementation)

Reply:
Added `UV_PYPY_INSTALL_MIRROR`.